lib/ansible/modules/dnf.py

```diff
@@ -523,9 +524,12 @@ def _compare_evr(self, e1, v1, r1, e2, v2, r2):
         return rc
 
     def _ensure_dnf(self):
-        if HAS_DNF:
+        best_parsable_locale = get_best_parsable_locale(self.module)
+        if HAS_DNF and os.environ.get('LC_ALL') == best_parsable_locale:
```

Comment:
@bcoca any thoughts as to better detection to prevent unnecessary respawns? Maybe checking if `LC_ALL` is one of https://github.com/ansible/ansible/blob/22aa6dedc9a69bb55f91ff95eab6b2844bf5d826/lib/ansible/module_utils/common/locale.py#L35?

Reply:
a new function in locale that sets only after comparing them to the 'prefered'?

Reply:
```
diff --git a/lib/ansible/module_utils/common/locale.py b/lib/ansible/module_utils/common/locale.py
index 56d8c23261..d0772024af 100644
--- a/lib/ansible/module_utils/common/locale.py
+++ b/lib/ansible/module_utils/common/locale.py
@@ -7,6 +7,11 @@ __metaclass__ = type
 from ansible.module_utils._text import to_native


+# new POSIX standard or English cause those are messages core team expects
+# yes, the last 2 are the same but some systems are weird
+PARSABLE_LOCALES =  ['C.utf8', 'en_US.utf8', 'C', 'POSIX']
+
+
 def get_best_parsable_locale(module, preferences=None):
     '''
         Attempts to return the best possible locale for parsing output in English
@@ -27,9 +32,7 @@ def get_best_parsable_locale(module, preferences=None):
     found = 'C'  # default posix, its ascii but always there

     if preferences is None:
-        # new POSIX standard or English cause those are messages core team expects
-        # yes, the last 2 are the same but some systems are weird
-        preferences = ['C.utf8', 'en_US.utf8', 'C', 'POSIX']
+        preferences = PARSABLE_LOCALES

     rc, out, err = module.run_command([locale, '-a'])

@@ -47,3 +50,19 @@ def get_best_parsable_locale(module, preferences=None):
                 found = pref
                 break
     return found
+
+
+def set_parsable_locale(best_locale, envvars=None, preferences=None):
+    '''
+    '''
+
+    if envvars is None:
+        envvars = ['LC_ALL', 'LANG']
+
+    if preferences is None:
+        preferences = PARSABLE_LOCALES
+
+    for envvar in envvars:
+        if os.environ.get(envar) in PARSABLE_LOCALES:
+            continue
+        os.environ[envvar] best_locale
```
something like this?

Reply:
In the end I went with postponing the imports to prevent respawns.